maplibre-native-ios/Amazon Location Service Demo/AWSSignatureV4Delegate.swift

```diff
@@ -4,67 +4,138 @@
+///
+/// - note: This is a workaround for [https://github.com/aws-samples/amazon-location-samples/issues/31](https://github.com/aws-samples/amazon-location-samples/issues/31).
+/// - seealso: https://github.com/mapbox/mapbox-gl-native/issues/12026 for an explanation of `NSURLProtocol`
+/// - seealso: https://github.com/mapbox/mapbox-gl-native/pull/13886/files for usage in the Mapbox library.
+class MapboxSigV4RequestInterceptor: URLProtocol {
+    lazy var session: URLSession = {
+        let config = URLSession.shared.configuration
+        return .init(configuration: config, delegate: self, delegateQueue: nil)
+    }()
+    
+    var dataTask: URLSessionDataTask?
+    private static var mapboxConfig = MapboxConfig()
+    private static var urlPrefix: String {
+        "https://maps.geo.\(Self.mapboxConfig.regionName).amazonaws.com"
+    }
+    
+    override class func canInit(with request: URLRequest) -> Bool {
+        return canInit(with: request.url)
+    }
+    
+    override class func canInit(with task: URLSessionTask) -> Bool {
+        return canInit(with: task.currentRequest?.url)
+    }
+    
+    override class func canonicalRequest(for request: URLRequest) -> URLRequest {
+        return request
+    }
+    
+    override func startLoading() {
+        guard let unsignedURL = request.url else {
+            return
+        }
+        
         // URL-encode spaces, etc.
-        let keyPath = String(url.path.dropFirst())
+        let keyPath = String(unsignedURL.path.dropFirst())
         guard let percentEncodedKeyPath = keyPath.addingPercentEncoding(withAllowedCharacters: .urlPathAllowed) else {
-            print("Invalid characters in path '\(keyPath)'; unsafe to sign")
-            return url
+            return assertionFailure("Invalid characters in path '\(keyPath)'; unsafe to sign")
         }
-
-        let endpoint = AWSEndpoint(region: region, serviceName: "geo", url: url)
-        let requestHeaders: [String: String] = ["host": endpoint!.hostName]
-
+        
+        guard let endpoint = AWSEndpoint(region: Self.mapboxConfig.region, serviceName: "geo", url: unsignedURL) else {
+            return assertionFailure("AWSEndpoint is nil")
+        }
+        let requestHeaders: [String: String] = ["host": endpoint.hostName]
+        
         // sign the URL
         let task = AWSSignatureV4Signer
             .generateQueryStringForSignatureV4(
-                withCredentialProvider: credentialsProvider,
+                withCredentialProvider: Self.mapboxConfig.credentialsProvider,
                 httpMethod: .GET,
                 expireDuration: 60,
-                endpoint: endpoint!,
+                endpoint: endpoint,
                 // workaround for https://github.com/aws-amplify/aws-sdk-ios/issues/3215
-                keyPath: AWSSignatureV4Delegate.doubleEncode(path: percentEncodedKeyPath),
+                keyPath: Self.doubleEncode(path: percentEncodedKeyPath),
```

Comment:
This is fixed in the next AWS iOS SDK release: https://github.com/aws-amplify/aws-sdk-ios/issues/3215#issuecomment-932630713

Once that's been released, we'll update the `Podfile` and remove the workarounds.

Reply:
Workaround removed in https://github.com/aws-samples/amazon-location-samples/pull/71

```suggestion
                keyPath: percentEncodedKeyPath,
```